qualtran/bloqs/basic_gates/z_basis.py

```diff
@@ -262,13 +283,67 @@ def as_cirq_op(
     def _t_complexity_(self) -> 'TComplexity':
         return TComplexity(clifford=1)
 
+    def wire_symbol(
+        self, reg: Optional['Register'], idx: Tuple[int, ...] = tuple()
+    ) -> 'WireSymbol':
+        if reg is None:
+            return Text('')
+
+        return TextBox('Z')
+
 
 @bloq_example
 def _zgate() -> ZGate:
     zgate = ZGate()
     return zgate
 
 
+@frozen
+class CZ(Bloq):
```

Comment:
https://github.com/quantumlib/Qualtran/pull/1115/files adds a `CYGate` whereas this PR adds a `CZ` gate. Let's keep the `CX / CY / CZ` family's name consistent -- let's either add `CPGate` or `CP` where `P` is the pauli in question.

Reply:
My original motivation was to use `CNOT` and `CZ` for those two gates since those are very common names. I.e. folks would expect to find a CNOT bloq and a CZ bloq through tab completion or searching the docs site. `CY` just doesn't feel as common to me, although I take the point on the consistency angle. But I really don't want to change `CNOT` to be `CX`.